File: impl/ts/tests/delete.test.ts
// DELETE: by predicate, no-WHERE (all rows), Kleene (NULL rows not matched), and the
// load-bearing no-PK rowid fix — DELETE then INSERT must not collide on a freed rowid.

import assert from "node:assert/strict";
import { test } from "node:test";
import { errCode } from "./util.ts";
import { memDb } from "./mem_db.ts";

test("delete from a missing table traps 42P01", () => {
  assert.equal(
    errCode(() => memDb().session().execute("DELETE FROM nope")),
    "42P01",
  );
});

File: impl/go/delete_test.go
package jed

// Step 6: DELETE — predicate-matched removal, no-WHERE clears, three-valued logic,
// and the no-PK monotonic-rowid regression (DELETE then INSERT must not collide).

import "testing"

func setupDelete(t *testing.T) *Session {
	return dbWith(
		t,
		"CREATE TABLE t (id i32 PRIMARY KEY, v i16)",
		"INSERT INTO t VALUES (1, 10)",
		"INSERT INTO t VALUES (2, 20)",
		"INSERT INTO t VALUES (3, 30)",
		"INSERT INTO t VALUES (4, NULL)",
	)
}

func TestDeleteMissingTable(t *testing.T) {
	wantErr(t, memDB().Session(SessionOptions{}), "DELETE FROM nope", "42P01")
}

func TestDeleteUnknownColumn(t *testing.T) {
	wantErr(t, setupDelete(t), "DELETE FROM t WHERE nope = 1", "42703")
}
